tests/test_order.py

```diff
@@ -1,106 +1,120 @@
-
-    url_order_info = f"{constants.API_HOST}/v2/orders/{constants.ORDER_ID}"
-    requests_mock.get(url=url_order_info, json=order_response)
-    url_asset_info = f"{constants.API_HOST}/v2/assets?search={constants.ORDER_ID}&size=50"
-    requests_mock.get(url=url_asset_info, json=fixtures_order.JSON_GET_ASSETS_RESPONSE)
-    order_placed = order.Order(auth=auth_mock, order_id=constants.ORDER_ID)
-    (asset_returned,) = order_placed.get_assets()
-    assert isinstance(asset_returned, asset.Asset)
-    assert asset_returned.asset_id == constants.ASSET_ORDER_ID
-
-
-@pytest.mark.parametrize(
-    "status",
-    [
-        "CREATED",
-        "BEING_PLACED",
-        "PLACED",
-        "PLACEMENT_FAILED",
-        "DELIVERY_INITIALIZATION_FAILED",
-        "BEING_FULFILLED",
-        "DOWNLOAD_FAILED",
-        "DOWNLOADED",
-        "FAILED_PERMANENTLY",
-    ],
-)
-def test_should_fail_to_get_assets_for_unfulfilled_order(auth_mock, requests_mock, status):
-    order_response = {"id": constants.ORDER_ID, "status": status}
-    url_order_info = f"{constants.API_HOST}/v2/orders/{constants.ORDER_ID}"
-    requests_mock.get(url=url_order_info, json=order_response)
-    order_placed = order.Order(auth=auth_mock, order_id=constants.ORDER_ID)
-    with pytest.raises(ValueError):
-        order_placed.get_assets()
+    def test_should_compute_is_fulfilled(self, auth_mock, requests_mock: req_mock.Mocker, info: dict, expected: dict):
+        requests_mock.get(url=ORDER_URL, json=info)
+        order_obj = order.Order(auth_mock, order_id=constants.ORDER_ID, order_info=info)
+        assert order_obj.is_fulfilled == expected
+
+    def test_should_get_assets_if_fulfilled(self, auth_mock, requests_mock):
+        info = {"id": constants.ORDER_ID, "status": "FULFILLED"}
+        requests_mock.get(url=ORDER_URL, json=info)
+        url_asset_info = f"{constants.API_HOST}/v2/assets?search={constants.ORDER_ID}"
+        asset_info = {
+            "content": [
+                {
+                    "id": constants.ASSET_ORDER_ID,
+                }
+            ],
+            "totalPages": 1,
+            "totalElements": 1,
+        }
+        requests_mock.get(url=url_asset_info, json=asset_info)
+        order_obj = order.Order(auth=auth_mock, order_id=constants.ORDER_ID)
+        assets = order_obj.get_assets()
+        assert all(isinstance(asset_element, asset.Asset) for asset_element in assets)
+        assert all(asset_element.asset_id == constants.ASSET_ORDER_ID for asset_element in assets)
+
+    @pytest.mark.parametrize(
+        "status",
+        [
+            "CREATED",
+            "BEING_PLACED",
+            "PLACED",
+            "PLACEMENT_FAILED",
+            "DELIVERY_INITIALIZATION_FAILED",
+            "BEING_FULFILLED",
+            "DOWNLOAD_FAILED",
+            "DOWNLOADED",
+            "FAILED_PERMANENTLY",
+        ],
+    )
+    def test_fails_to_get_assets_if_not_fulfilled(self, status):
+        auth = mock.MagicMock()
```

Comment:
We need real auth_mock here, since the status is queried eagerly

Reply:
But the fail error is returned before actually call the endpoint.

Reply:
No, it calls the mock and fails because the mock still doesn't return the correct status (but not the one we expect it to return)